### DISCUSSION TITLE
implement a set of cheesy post-install system modification scripts.

### DISCUSSION THREAD

**Discussion by zcot** [linuxmint/discussions#5] [Ideas]
The set of total-system-modification scripts should probably have a category name. "cheesy scripts" is maybe not a great start, but it could probably be worse!

I often see users who will want a particular niche setup for some specific reason.

Some have really low hardware so they are looking to "speed it up"(obviously maybe choose a really low resource DE, but even on Xfce the opportunity is there). So, ok it's old, there's no wifi? there's no bluetooth? I don't even know what a printer is? Let us just back out of these things.

Another case, somebody has very low disk space, a simple script could eliminate some base program installations, maybe adjust the logging parameters for a quite minimal analysis, or it could even be two separated system-altering scripts. Obviously for the experienced users we probably see "saving space" is only relating to maybe a couple GB if we really go hard at it, but at least it's something. And another script, we have some good guides on the forums, "free up some space", where we get into the basic autoremoval, cache clearing, etc. I suspect there could be a multiple choice setup and it wouldn't be too hard to show actual disk space data comparison.

Here's another, a user wants the benefits of the Ubuntu Studio setup, but they really want Mint, so the cheesy script would be able to make a one-step process, throw a real-time kernel at it, and add the choice of audio, video, image, or whatever need they would desire from a "studio" setup. I imagine it even as some inexpensive bash scripts... but they could have a little bit of color and some occasional ascii graphics too, to make it a little more cheesy.

I saw someone recently who wanted to remove libreoffice from Mint 21.x and there was no metapackage installation so the removal would be more of a complexity, possibly considered intermediate or advanced(for a newbie)? Of course a script could be provided to handle that, or actually that situation doesn't especially fit into the idea I am thinking about here, probably an according non-ubuntu metapackage, but it's not especially too far out of the line of being able to have a set of scripts that can "do stuff" with an easy invocation.

Some of the guys on IRC mention having "th3 hax0r" interface, so there's an opportunity to have a switching script for a few different PS1 options, of course 'default' would be one as well.

Does it need to have some global file to specify whether some cheesy thing has been implemented or not, perhaps useful in order to unwind it?

Obviously these types of ideas can all be done manually, and are not even especially hard, perhaps only a little bit time-consuming, for those with some experience ready to tackle the topics directly.

We have the Cinnamon Spices...

Is it odd to think we can get some cheese on it?

**Comment by clefebvre** [comment]
People have so many different niche use case... and not only that, most of them aren't techie enough to run scripts, let alone adapt them. Presenting a search engine of scripts could be achieved with a bot on IRC or Discord, or a collection of Gists..

Now, when you talk about Saving HDD space though. That's a use case which is very common and affects many users. This is something that needs to be tackled and integrated. Right now we've some level of integration in the DE itself, but to go further with this we need to implement a proper tool. We'll make an app for that in the near future. That's in the roadmap already.

Another thing we have in mind are nemo actions in cinnamon spices. We have that eco-system for themes, applets, desklets, extensions, we'd extend it to nemo actions.. so people could search and install right-click actions and easily find niche functionality. Split this MKV into smaller files, make a thumbnail of that image, etc etc... Note that even this is a little bit niche, but at least thanks to the spice ecosystem and the fact that it's easy to use (through nemo), it would get some level of visibility.
